src/group_theory/complement.lean

```diff
@@ -406,3 +407,76 @@ begin
 end
 
 end subgroup
+
+namespace subgroup
+
+open equiv function mem_left_transversals mul_action mul_action.quotient zmod
+
+universe u
+
+variables {G : Type u} [group G] (H : subgroup G) (g : G)
+
+/-- Partition `G ⧸ H` into orbits of the action of `g : G`. -/
+noncomputable def quotient_equiv_sigma_zmod : G ⧸ H ≃
+  Σ (q : orbit_rel.quotient (zpowers g) (G ⧸ H)), zmod (minimal_period ((•) g) q.out') :=
+(self_equiv_sigma_orbits (zpowers g) (G ⧸ H)).trans
+  (sigma_congr_right (λ q, orbit_zpowers_equiv g q.out'))
+
+lemma quotient_equiv_sigma_zmod_symm_apply
+  (q : orbit_rel.quotient (zpowers g) (G ⧸ H)) (k : zmod (minimal_period ((•) g) q.out')) :
+  (quotient_equiv_sigma_zmod H g).symm ⟨q, k⟩ = g ^ (k : ℤ) • q.out' :=
+rfl
+
+lemma quotient_equiv_sigma_zmod_apply (q : orbit_rel.quotient (zpowers g) (G ⧸ H)) (k : ℤ) :
+  quotient_equiv_sigma_zmod H g (g ^ k • q.out') = ⟨q, k⟩ :=
+by rw [apply_eq_iff_eq_symm_apply, quotient_equiv_sigma_zmod_symm_apply,
+  zmod.coe_int_cast, zpow_smul_mod_minimal_period]
+
+/-- The transfer transversal as a function. -/
+noncomputable def transfer_function : G ⧸ H → G :=
+λ q, g ^ ((quotient_equiv_sigma_zmod H g q).2 : ℤ) * (quotient_equiv_sigma_zmod H g q).1.out'.out'
+
+lemma transfer_function_apply (q : G ⧸ H) : transfer_function H g q =
+  g ^ ((quotient_equiv_sigma_zmod H g q).2 : ℤ) * (quotient_equiv_sigma_zmod H g q).1.out'.out' :=
+rfl
+
+lemma coe_transfer_function (q : G ⧸ H) : ↑(transfer_function H g q) = q :=
+by rw [transfer_function_apply, ←smul_eq_mul, coe_smul_out',
+  ←quotient_equiv_sigma_zmod_symm_apply, sigma.eta, symm_apply_apply]
+
+/-- The transfer transversal as a set. -/
+def transfer_set : set G :=
+set.range (transfer_function H g)
+
+lemma mem_transfer_set (q : G ⧸ H) : transfer_function H g q ∈ transfer_set H g :=
+⟨q, rfl⟩
+
+/-- The transfer transversal. -/
```

Comment:
Could you please expand these docstrings a bit more? Try to write a helpful sentence or two for readers who do not yet know the theory by heart.

Reply:
Sure, I've added a bit more description of these objects.